api/src/main/java/org/pmiops/workbench/reporting/ReportingVerificationServiceImpl.java

```diff
@@ -0,0 +1,154 @@
+package org.pmiops.workbench.reporting;
+
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.QueryParameterValue;
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.ImmutableList;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.StreamSupport;
+import javax.inject.Provider;
+import org.pmiops.workbench.api.BigQueryService;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.model.ReportingSnapshot;
+import org.pmiops.workbench.model.ReportingUploadDetails;
+import org.pmiops.workbench.model.ReportingUploadResult;
+import org.pmiops.workbench.reporting.insertion.CohortColumnValueExtractor;
+import org.pmiops.workbench.reporting.insertion.InstitutionColumnValueExtractor;
+import org.pmiops.workbench.reporting.insertion.UserColumnValueExtractor;
+import org.pmiops.workbench.reporting.insertion.WorkspaceColumnValueExtractor;
+import org.pmiops.workbench.utils.FieldValues;
+import org.pmiops.workbench.utils.LogFormatters;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ReportingVerificationServiceImpl implements ReportingVerificationService {
+  private static final Logger logger =
+      Logger.getLogger(ReportingVerificationServiceImpl.class.getName());
+
+  private final BigQueryService bigQueryService;
+  private final Provider<WorkbenchConfig> workbenchConfigProvider;
+  private Provider<Stopwatch> stopwatchProvider;
+
+  public ReportingVerificationServiceImpl(
+      BigQueryService bigQueryService,
+      Provider<WorkbenchConfig> workbenchConfigProvider,
+      Provider<Stopwatch> stopwatchProvider) {
+    this.bigQueryService = bigQueryService;
+    this.workbenchConfigProvider = workbenchConfigProvider;
+    this.stopwatchProvider = stopwatchProvider;
+  }
+
+  @Override
+  public ReportingUploadDetails checkUploads(ReportingSnapshot snapshot) {
```

Comment:
This isn't checking anything, it's just building a `ReportingUploadDetails`, perhaps the function name could reflect that